src/dataflow/src/render/reduce.rs

```diff
@@ -502,12 +508,66 @@ impl ReducePlan {
                     ));
                 }
                 // Now we need to collate them together.
-                build_collation(to_collate, expr.aggregate_types, &mut collection.scope())
+                build_collation(to_collate, expr.aggregate_types, &mut collection.scope()).into()
             }
+        };
+        arrangement_or_bundle.to_bundle(key_arity, err_input)
+    }
+}
+
+enum ArrangementOrBundle<G, T>
+where
+    G: Scope,
+    G::Timestamp: Lattice + Refines<T>,
+    T: Timestamp + Lattice,
+{
+    Arrangement(Arrangement<G, Row>),
+    CollectionBundle(CollectionBundle<G, Row, T>),
+}
+
+impl<G, T> ArrangementOrBundle<G, T>
+where
+    G: Scope,
+    G::Timestamp: Lattice + Refines<T>,
+    T: Timestamp + Lattice,
+{
+    fn to_bundle(
+        self,
+        key_arity: usize,
+        err_input: Collection<G, DataflowError>,
```

Comment:
This is a bit weird that for one of the variants it ignores the `err_input` argument. I worry that it will result in some future bug where someone imagines that it will concat that input in rather than drop it when you have an -`Arrangement`.

Reply:
Fixed, `err_input` is now used for both variants.